DuplicatesFinder_v4/ViewModels/MainViewModel.cs

```diff
@@ -151,7 +153,7 @@ public ICommand OnClickExport
             {
                 return onClickExport ?? (onClickExport = new RelayCommand((r) =>
                 {
-                    Task.Run(() => DuplicatesViewModel.SaveToTxt());
+                    Task.Run(() => DuplicatesViewModel.SaveToTxtAsync());
                      
                     MessageBoxResult result = System.Windows.MessageBox.Show(

```

Comment:
not very correct.
Above you called async method without waiting.
This means that you will show messagebox before that method will complete

You need to somehow show this messagebox after task above will complete (maybe pass some lambda to called method)

Reply:
Tnx. Fixed. Used lambda.